src/gatsby-node.js

```diff
@@ -248,3 +250,37 @@ exports.createResolvers = ({
   });
   createResolvers(resolvers);
 };
+
+exports.pluginOptionsSchema = ({ Joi }) => {
+  return Joi.object({
+    api_key: Joi.string().required().description(`API Key is a unique key assigned to each stack.`),
+    delivery_token: Joi.string().required().description(`Delivery Token is a read-only credential.`),
+    environment: Joi.string().required().description(`Environment where you published your data.`),
+    cdn: Joi.string().default(`https://cdn.contentstack.io/v3`).description(`CDN set this to point to other cdn end point. For eg: https://eu-cdn.contentstack.com/v3 `),
+    type_prefix:  Joi.string().default(`Contentstack`).description(`Specify a different prefix for types. This is useful in cases where you have multiple instances of the plugin to be connected to different stacks.`),
+    expediteBuild: Joi.boolean().default(false).description(`expediteBuild set this to either true or false.`),
+    enableSchemaGeneration: Joi.boolean().default(false).description(`Specify true if you want to generate custom schema.`),
+  }).external(validateContentstackAccess)
+}
+
+
+const validateContentstackAccess = async pluginOptions => {
+  if (process.env.NODE_ENV === `test`) return undefined
+  
+  let host = pluginOptions.cdn ? pluginOptions.cdn : 'https://cdn.contentstack.io/v3';
+  await fetch(`${host}/content_types?include_count=false`, {
+    headers: {
+      "api_key" : `${pluginOptions.api_key}`,
+      "access_token": `${pluginOptions.delivery_token}`,
+    },
+  })
+    .then(res => res.ok)
```

Comment:
```
.then(res => {
    if (!res.ok) {
      throw new Error(
          `Cannot access Contentstack with api_key=${pluginOptions.api_key} & delivery_token=${pluginOptions.delivery_token}.`
        )
    }
})
```


No need of adding another .then as res.ok is a boolean property and can be accessed there itself.